### PR TITLE
Adds a new form and widget that implements CC and BCC fields for all emails

### LINKED ISSUE #2086: Add CC, BCC and Attachment field to all emails
It would be helpful to allow arbitrary CCs on all emails for those who want them.

- [ ]  All email views have a CC input

- [ ] All email views have a BCC input

- [ ] All email views have an attachment input

### PR DESCRIPTION
**Pull request by mauromsl**
closes #2086 
Since this PR touches a lot of files across all apps I have organised the commits in this large working branch into two blocks to facilitate code review:

**New email forms:**

https://github.com/BirkbeckCTP/janeway/pull/3121/files/5935df6275905a6108c527a59ff1f62979929236..b0f04361229505cb27430a91b0e7c5105c64a271

**Use of those forms across the workflow:**
https://github.com/BirkbeckCTP/janeway/pull/3121/files/b0f04361229505cb27430a91b0e7c5105c64a271..8fc2ea1216d6300067ed9b1aaaceeef04b955f9f

### CODE REVIEW

File: src/core/forms/forms.py [hunk 1]
```diff
@@ -748,3 +749,70 @@ def is_confirmed(self):
...
+    matching the given setting_name
+    :param setting_name: The name of the setting (Group must be email)
+    :param email_context: A dict of the context required to populate the email
+    :param request: The instance of this HttpRequest
+    :param journal: (Optional) an instance of journal.models.Journal
+    """
+    def __init__(self, *args, **kwargs):
+        setting_name = kwargs.pop("setting_name")
+        email_context = kwargs.pop("email_context", {})
+        subject_setting_name = "subject_" + setting_name
```
Comment (joemull) [inline src/core/forms/forms.py:804]: This assumes that subject settings are always named exactly after their corresponding email settings. Have you checked whether that is a safe assumption?

**Review by joemull — APPROVED** [review]
This is a lot of work, and I really appreciate that it establishes type-safe forms and fields that are re-usable for every email interface. In that sense, it's a massive step in the right direction.

As for how to test it, to what extent have you manually tested it? I am a bit unsure how to go about that. 

I added a couple of comments but I don't know how actionable they are.

Overall, I'm guessing this work introduces a few bugs because of how much code it touches, but I'm approving it as I think it's worth moving ahead and figuring out the exceptions as they arise.

**Comment by ajrbyers** [comment]
I was just looking over this... it looks like it updates review but none of the other workflow elements, is that correct?

**Review by ajrbyers — CHANGES_REQUESTED** [review]

**Comment by mauromsl** [comment]
> This is a lot of work, and I really appreciate that it establishes type-safe forms and fields that are re-usable for every email interface. In that sense, it's a massive step in the right direction.
> 
> As for how to test it, to what extent have you manually tested it? I am a bit unsure how to go about that.
> 
> I added a couple of comments but I don't know how actionable they are.
> 
> Overall, I'm guessing this work introduces a few bugs because of how much code it touches, but I'm approving it as I think it's worth moving ahead and figuring out the exceptions as they arise.

I've tweaked the CSS so that at least the field looks the same as any other inputs.
In terms of accessibility, I think our Admin UI is due a whole revamp (including the tagging library) it is not something that pertains this one issue.
image

**Review by ajrbyers — APPROVED** [review]

**Comment by ajrbyers** [comment]
@mauromsl this looks good, but there are 139 failed tests....

**Comment by mauromsl** [comment]
> @mauromsl this looks good, but there are 139 failed tests....

yeah, the merge with master was missing a tweak required after a refactor. I've also fixed the test data on unit tests that were broken due to my changes. I think the others were already failing on master due to various issues with merged code

**Pull request merged**